src/object.d

```diff
@@ -1191,6 +1211,38 @@ class TypeInfo_Interface : TypeInfo
     override @property uint flags() nothrow pure const { return 1; }
 
     TypeInfo_Class info;
+
+   /**
+    * Returns true if the class described by `child` derives from the
+    * interface described by this `TypeInfo_Interface`.
+    *
+    * Params:
+    *  child = TypeInfo for some class
+    * Returns:
+    *  true if the class described by `child` derives from the
+    *  interface described by this `TypeInfo_Interface`.
+    */
+    bool isBaseOf(scope const TypeInfo_Class child) const @nogc nothrow pure @trusted
```

Comment:
Maybe this and the other functions should be `final`. The limitation could be lifted later if necessary.

Reply:
Sounds good to me.